invenio_search/ext.py

```diff
@@ -332,14 +338,15 @@ def _build(tree_or_filename, alias=None):
                         ))
             if alias:
                 alias_indices = self._get_indices(tree_or_filename)
-                alias_indices = [new_indices[i] for i in alias_indices]
-                alias_name = build_alias_name(alias, app=self.app)
-                ensure_not_exists(alias_name)
-                actions.append(dict(
-                    type='create_alias',
-                    index=alias_indices,
-                    alias=alias_name
-                ))
+                alias_indices = [new_indices.get(i) for i in alias_indices]
+                if alias_indices and None not in alias_indices:
```

Comment:
The `None not in alias_indices` sounds a bit weird. Maybe it's cleaner to filter the list above:

```python
alias_indices = [new_indices[i] for i in alias_indices if i in new_indices]
```

Reply:
This list shouldn't be filtered. It just should be skipped if there is any `None` inside as it means that this index is already created. So it can be like that `None not in alias_indices` or additional if which checks if name exists before every `actions.append`

Reply:
I'll describe the use case to give you a proper example how I think of it.

You have a mapping hierarchy like so:

- records
  - article
  - dataset
  - video

*(For simplification no prefixing is configured)*

```python
# Let's assume that you only want to create the `records-video` index:
current_search.create(index_list=['records-video'])

# This will now only go through the creation operations for the `records-video` 
# index, and thus add it to the `new_indices` dictionary with its respective 
# prefixed + suffixed index name.
assert new_indices == {'records-video': 'records-video-1234567'}  # timestamp prefix

# When you reach the point where you have to create the `records` alias,:
alias_indices = self._get_indices(tree_or_filename)
# "self._get_indices" doesn't take "index_list" into consideration
assert alias_indices == ['records-article', 'records-dataset', 'records-video']

# Then you basically want to figure out the final index names 
# (with prefix/suffix) in order to be able to properly update the alias
alias_indices = [new_indices.get(i) for i in alias_indices]
assert alias_indices == [None, None, 'records-video-1234567']

# Now if you just check if `None` is in the "alias_indices" list, you will
# bail out on adding the newly created index to the "records" alias

# Instead IMHO, the logic should be that you definitely want to create aliases
# for the index you've just created.
```

Since this has become quite lengthy, I'm more than happy to discuss this in-chat (I guess the Mattermost channel would be the most appropriate place) or IRL, since I am aware that the changes introduced in v1.2.0 were pretty big and the plan was to make e.g. things like index migrations easier (that's also the reason `current_search.create_index` was introduced) and not break people's repositories :) (@drjova and @benjamin-bergia feel free of course to join as well)

Reply:
Ok, I get it now. Sorry, you were right. I updated code and test.